crates/wadm-types/src/lib.rs

```diff
@@ -67,11 +69,65 @@ impl Manifest {
             .map(|v| v.as_str())
     }
 
+    /// Indicates if the manifest is shared, meaning it can be used by multiple applications
+    pub fn shared(&self) -> bool {
+        self.metadata
+            .annotations
+            .get(SHARED_ANNOTATION_KEY)
+            .is_some_and(|v| v.parse::<bool>().unwrap_or(false))
+    }
+
     /// Returns the components in the manifest
     pub fn components(&self) -> impl Iterator<Item = &Component> {
         self.spec.components.iter()
     }
 
+    /// Helper function to find shared components that are missing from the given list of
+    /// deployed applications
+    pub fn missing_shared_components(&self, deployed_apps: &Vec<&Manifest>) -> Vec<&Component> {
```

Comment:
Nit: This should probably just be `&[&Manifest]` instead. Generally clippy warns on this too

Reply:
This gave me some issue w/ collecting into a Sized collection /shrug